src/twisted/application/twist/test/test_twist.py

```diff
@@ -191,3 +191,80 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    """
+    Tests to verify that the Twist script takes the expected actions related
+    to signals and the reactor.
+    """
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def fakeExitWithSignal(sig):
+            """
+            Fake to capture whether L{twisted.application._exitWithSignal
+            was called.
+
+            @param sig: Signal value
+            @type sig: C{int}
+            """
+            self.exitWithSignalCalled = True
+
+        self.patch(_twist, '_exitWithSignal', fakeExitWithSignal)
+
+        def startLogging(_):
+            """
+            Prevent Runner from adding new log observers or other
+            tests outside this module will fail.
+
+            @param _: Unused self param
+            """
+            pass
+
+        self.patch(Runner, 'startLogging', startLogging)
```

Comment:
Specifically, one of the errors when a dummy logger instance is passed is:
```
twisted.trial.unittest.FailTest: 4 != 1

twisted.test.test_log.LogTests.test_erroneousErrors
```
I'll look into what the failing test are expecting, but I suspect removing the created observers in my test will do the trick.